### PR TITLE
[server] Single source of configuration I/II

### LINKED ISSUE #4650: [server] rewrite configuration to use a single config source
currently the server component's config consists of a number of env vars (which might be rendered from secrets) and mounts.

as suggested by @csweichel, it would be best to rewrite the configuration of the server component (perhaps payment as well, as they share some of the configuration,) to use a single source. advantages expected by this change are:
 * make telepresence easier
 * reduce turnarounds when working on changes related to payment or github

### PR DESCRIPTION
**Pull request by geropl**
Attempts to fix #4650. This is the 1st in a series of 2 PRs.

It turns out the helm chart and parsing interaction in `Env` is rather complicated. To make sure we do not miss any variance we take a 2 step approach:
 1. Introduce new `Config` shape which contains everything from `Env` (minus some basic refactoring and cleanup). This is **not used yet** to make sure we don't break anything, but instead just printed out on startup for reference.
 2. The next step is to ease the process of serializing->deserializing the config from helm chart to `Config`, ideally reducing it to a `{{ toJson | b64enc }}` on helm followed by a `JSON.parse(...)` + validation in `server`. To make sure we don't break stuff we can compare the result to the print-outs from `staging` and `devstaging`
 
 For this PR, feedback is very welcome on:
  - the structure of `Config`
  - verification that the remaining unused fields in `Env` can be removed.

The rest is straight forward extract/removal.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4882#" title="Author self-approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/config.ts
Comment (AlexTugarev) [inline components/server/src/config.ts:73]: I guess so. 
The single usage I've seen recently was the sanity check for returnTo URLs where we want to allow redirects to the configured `homepage`, which in turn can be remodeled as config attribute in here.

**Review by Goldselleruk — APPROVED** [review]
#Gdzie gold

**Comment by roboquat** [comment]
@Goldselleruk: changing LGTM is restricted to collaborators

<details>

In response to [this](https://github.com/gitpod-io/gitpod/pull/4882#pullrequestreview-714813051):

> #Gdzie gold 


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4882#" title="Author self-approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4882#pullrequestreview-714813051" title="Approved">Goldselleruk</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by AlexTugarev — COMMENTED** [review]
Code changes look good. Thanks @geropl!
Let's resolve the merge conflicts and perhaps wait with merging for todays deployment in order to verify on staging one more time.

File: components/server/ee/src/config.ts
Comment (geropl) [reply components/server/ee/src/config.ts:11]: Yes, was thinking about the same, but wanted to confirm with you. Will do!

**Comment by geropl** [comment]
@AlexTugarev I rebased, merged Env/EnvEE -> Config and squashed commits: ready to merge!

File: components/server/src/server.ts
Comment (AlexTugarev) [inline components/server/src/server.ts:85]: 👍🏻

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 7f2752ee0e556fad6c7ea68ed6e637e559627209</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4882#issuecomment-887339334" title="LGTM">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4882#" title="Author self-approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4882#pullrequestreview-714813051" title="Approved">Goldselleruk</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [AlexTugarev,geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev,geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
@geropl feel free to release the break ;-)

**Pull request merged**
